great_expectations/rule_based_profiler/parameter_builder/histogram_parameter_builder.py

```diff
@@ -26,11 +26,22 @@
 )
 
 
-class PartitionParameterBuilder(MetricSingleBatchParameterBuilder):
+class HistogramParameterBuilder(MetricSingleBatchParameterBuilder):
     """
     Compute histogram/partition using specified metric (depending on bucketizaiton directive) for one Batch of data.
+
+    The histogram parameter builder handles building histograms from both continuous and discrete distributions.
+    A discrete distribution will be produced when `bucketize_data` is set to False.
+
+    If `bucketize_data` is set to true, then the parameter builder will first determine bins using the method
+    specified by the `bins` argument, and then compute the histogram using those bins.
     """
 
+    # NOTE - 20220823 - JPC - I believe supporting explicitly-provided bins could also make sense, but
+    # that it is not the dominant case for profiling data.
+    # Allowing that would align with the numpy.histogram api. However, it would overload the definition
```

Comment:
@jcampbell One more quick thought: I feel that we are already overloading the semantics of `bins` in `ColumnPartition`: the input comes in as type `str` ("uniform", "ntile", "quantile", "percentile", "auto"), while the output is a `List` (e.g., generated by `np.linspace()` in the case of floating point valued bins).  Should we do anything about this?  Thanks.

Reply:
I think we're adhering to the numpy api, so we shouldn't change it yet, but I am open to a change, yes.